moshi/src/test/java/com/squareup/moshi/MapJsonAdapterTest.java

```diff
@@ -141,6 +142,84 @@
     assertThat(jsonAdapter.fromJsonValue(jsonObject)).isEqualTo(map);
   }
 
+  @Test public void booleanKeyTypeHasCoherentErrorMessage() {
+    Map<Boolean, String> map = new LinkedHashMap<>();
+    map.put(true, "");
+    JsonAdapter<Map<Boolean, String>> adapter = mapAdapter(Boolean.class, String.class);
+    try {
+      adapter.toJson(map);
+      fail();
+    } catch (IllegalStateException expected) {
+      assertThat(expected).hasMessage("Boolean cannot be used as a map key in JSON.");
```

Comment:
tiny thing: can't recall if we prefer `expected` or `e` when matching error messages instead of ignoring the error.